core/src/test/java/org/elasticsearch/node/internal/InternalSettingsPreparerTests.java

```diff
@@ -42,6 +44,8 @@
 import static org.hamcrest.Matchers.*;
 
 public class InternalSettingsPreparerTests extends ESTestCase {
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
```

Comment:
Here would be something of an alternative maybe for the future, when java 8 is minimal (it would be less annoying due to effectively final and lambda):

    public void test() {
        assertError(IndexOutOfBoundsException.class, () -> {
            int foo[] = new int[5];
            System.out.println(foo[6]);
        });
    }

    // test helper for expected exception
    // TODO: can we replace Runnable with TestMethod or similar interface that throws Throwable
    // so checked exceptions arent annoying?
    static void assertError(Class<? extends Throwable> expectedClazz, Runnable foo) {
        assertError(expectedClazz, null, foo);
    }
    
    // test helper for expected exception, with expected message
    static void assertError(Class<? extends Throwable> expectedClazz, String expectedMessage, Runnable foo) {
        try {
            foo.run();
            fail("didnt hit expected exception, expected: " + expectedClazz.getSimpleName());
        } catch (Throwable t) {
            if (!expectedClazz.isAssignableFrom(t.getClass())) {
                throw new IllegalStateException("got the wrong exception, expected: " + expectedClazz.getSimpleName() + ", got: " + t, t);
            }
            if (t.getMessage() == null && expectedMessage != null) {
                throw new IllegalStateException("exception had null message, but expected: " + expectedMessage, t);
            }
            if (expectedMessage != null && !t.getMessage().contains(expectedMessage)) {
                throw new IllegalStateException("expection message did not contain expected text: " + expectedMessage, t);
            }
        }
    }